test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/main.py

```diff
@@ -271,6 +269,86 @@ def validate(self):
+            'spawnlpe': 'os.spawnlpe',
+            'spawnv': 'os.spawnv',
+            'spawnve': 'os.spawnve',
+            'spawnvp': 'os.spawnvp',
+            'spawnvpe': 'os.spawnvpe',
+            'system': 'os.system',
+        }
+        self.subprocess_methods = {
+            'Popen': 'subprocess.Popen',
+            'call': 'subprocess.call',
+            'run': 'subprocess.run',
+            'getstatusoutput': 'subprocess.getstatusoutput',
+            'getoutput': 'subprocess.getoutput',
+            'check_output': 'subprocess.check_output',
+            'check_call': 'subprocess.check_call',
+        }
+
+    def visit_ImportFrom(self, node):
+        """Checks for 'from something import anything' or 'from something import anything as mysys'."""
+        if node.module == 'os':
+            for alias in node.names:
+                # If aliased, store the alias name; otherwise, store particular name from import
+                if alias.name in self.os_methods:
+                    imported_name = alias.asname if alias.asname else alias.name
+                    self.import_names.append(imported_name)
+        elif node.module == 'subprocess':
+            for alias in node.names:
+                # If aliased, store the alias name; otherwise, store particular name from import
+                if alias.name in self.subprocess_methods:
+                    imported_name = alias.asname if alias.asname else alias.name
+                    self.import_names.append(imported_name)
+        self.generic_visit(node)
+
+    def visit_Call(self, node):
+        """Checks for function calls: os.system(...) or system(...) (if imported)."""
+
+        # Case 1: Checking imports
+        if isinstance(node.func, ast.Attribute):
+            if isinstance(node.func.value, ast.Name):
+                if node.func.value.id == 'os':
+                    if node.func.attr in self.os_methods:
+                        self.system_calls.append(
+                            (
+                                node.lineno, node.col_offset, self.os_methods.get(node.func.attr),
+                                f"use-run-command-not-os-{node.func.attr}"
+                            )
+                        )
+                elif node.func.value.id == 'subprocess':
+                    if node.func.attr in self.subprocess_methods:
+                        self.system_calls.append(
+                            (
+                                node.lineno, node.col_offset, self.subprocess_methods.get(node.func.attr),
+                                f"use-run-command-not-{node.func.attr.lower()}"
+                            )
+                        )
+
+        # Case 2: Checking for function calls
+        elif isinstance(node.func, ast.Name):
+            func_name = node.func.id
+            if func_name in self.import_names:
+                self.system_calls.append((node.lineno, node.col_offset, func_name, f"use-run-command-not-{func_name}"))
```

Comment:
@Akasurde thanks for the good work here so far! :-)

Personal preference: instead of creating (around of) 20 new different ignore directives, not to mention the fact that some of them, contrary to all other ignore directives, have `_` in the name, instead of `-` (I know, `.replace('_', '-')` could solve _that_ quickly, but nonetheless), I think it would be better to have one ignore directive for this one check.

Maybe just `use-run-command`? `use-run-command-instead`? Or `use-run-command-not-native-function`? The more specific the wording is, the longer and cumbersome it gets. 

Anyways, just my $0.02